crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,765 @@
+                RdpdrPdu::DeviceControlResponse(DeviceControlResponse {
+                    device_io_reply: DeviceIoResponse::new(req_inner.header, NtStatus::SUCCESS),
+                    output_buffer: None,
+                }),
+            )]),
+            ServerDriveIoRequest::ServerDriveLockControlRequest(_) => {
+                // TODO
+                Ok(Vec::new())
+            }
+        }
+    }
+}
+
+pub(crate) fn write_device(
+    backend: &mut SimpleRdpdrBackend,
+    req_inner: DeviceWriteRequest,
+) -> PduResult<Vec<SvcMessage>> {
+    let write_inner = |file: &mut std::fs::File, offset, write_data| -> std::io::Result<usize> {
+        let sf = SeekFrom::Start(offset);
+        file.seek(sf)?;
+        let length = file.write(write_data)?;
+        file.flush()?;
+        Ok(length)
+    };
+    process_dependent_file(
+        backend,
+        req_inner.device_io_request,
+        |request| {
+            let res = RdpdrPdu::DeviceWriteResponse(DeviceWriteResponse {
+                device_io_reply: DeviceIoResponse::new(request, NtStatus::NO_SUCH_FILE),
+                length: 0_u32,
+            });
+            Ok(vec![SvcMessage::from(res)])
+        },
+        |file, request| match write_inner(file, req_inner.offset, &req_inner.write_data) {
+            Ok(length) => {
+                if length == req_inner.write_data.len() {
+                    Ok(vec![SvcMessage::from(RdpdrPdu::DeviceWriteResponse(
+                        DeviceWriteResponse {
+                            device_io_reply: DeviceIoResponse::new(request, NtStatus::SUCCESS),
+                            length: req_inner.write_data.len() as u32,
+                        },
+                    ))])
+                } else {
+                    warn!("write len:{} is not equal to {}", length, req_inner.write_data.len());
+                    let res = RdpdrPdu::DeviceWriteResponse(DeviceWriteResponse {
+                        device_io_reply: DeviceIoResponse::new(request, NtStatus::UNSUCCESSFUL),
+                        length: 0_u32,
+                    });
+                    Ok(vec![SvcMessage::from(res)])
+                }
+            }
+            Err(e) => {
+                warn!("write error:{:?}", e);
+                let res = RdpdrPdu::DeviceWriteResponse(DeviceWriteResponse {
+                    device_io_reply: DeviceIoResponse::new(request, NtStatus::UNSUCCESSFUL),
+                    length: 0_u32,
+                });
+                Ok(vec![SvcMessage::from(res)])
+            }
+        },
+    )
+}
+pub(crate) fn read_device(
+    backend: &mut SimpleRdpdrBackend,
+    req_inner: DeviceReadRequest,
+) -> PduResult<Vec<SvcMessage>> {
+    let read_inner = |file: &mut std::fs::File, offset, length| -> std::io::Result<Vec<u8>> {
+        let sf = SeekFrom::Start(offset);
+        file.seek(sf)?;
+        let mut buf = vec![0; length];
+        use std::io::Read;
```

Comment:
suggestion: I’m all for putting local `use` inside functions, but this should be at the top of the outermost function for clarity.

```rust
fn read_device() {
    use std::io::Read as _;

    fn helper() {
        // Use std::io::Read
    }
}
```